policy/data/cache/cache.go

```diff
@@ -0,0 +1,52 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+)
+
+type QueryCache interface {
+	Read(ctx context.Context, query string, variables map[string]interface{}) ([]byte, error)
+	Write(ctx context.Context, query string, variables map[string]interface{}, res []byte) error
+}
+
+type cacheKey struct {
+	query     string
+	variables string
+}
+
+type SimpleQueryCache struct {
+	cache map[cacheKey][]byte
+}
+
+func NewQueryCache() SimpleQueryCache {
+	return SimpleQueryCache{
+		cache: make(map[cacheKey][]byte),
+	}
+}
+
+func (d SimpleQueryCache) Read(ctx context.Context, query string, variables map[string]interface{}) ([]byte, error) {
```

Comment:
Missing a mutex when reading from the cache?

Reply:
Yeah could do. The only place this would currently be used is not multi-threaded at the moment, but it's a fair point that it could be eventually.